### PR TITLE
feat(kadena-cli): add kadena-cli devnet commands

### PR DESCRIPTION
**Pull request by jessevanmuijden**
```
Usage: kadena-cli devnet [options] [command]

Tool for configuring, starting, stopping and managing local devnet containers

Options:
  -h, --help        display help for command

Commands:
  list              List all available devnet container configurations
  manage            Manage devnet(s)
  create [options]  Create new devnet
  run [options]     Run devnet
  stop [options]    Stop devnet
  remove [options]  Remove devnet container and configuration
  update [options]  Update devnet container image
  help [command]    display help for command
  ```

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205794789667220

### CODE REVIEW

File: packages/tools/kadena-cli/src/utils/globalOptions.ts [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+  }),
+  devnetUseVolume: createOption({
+    key: 'useVolume' as const,
+    prompt: devnetUseVolumePrompt,
+    validation: z.boolean(),
+    option: new Option(
+      '-u, --useVolume',
+      'Create a persistent volume to mount to the container'
+    ),
+  }),
```
Comment (alber70g) [inline packages/tools/kadena-cli/src/utils/globalOptions.ts:142]: Should we parse this into a boolean, or would that interfere with the prompts

`new Option().argParser(Boolean)`

File: packages/tools/kadena-cli/src/account/accountHelpers.ts [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+  const accountFilePath = path.join(
+    defaultAccountsPath,
+    `${sanitizedName}.yaml`,
+  );
+
+  writeFile(
+    accountFilePath,
+    yaml.dump(options),
+    'utf8' as WriteFileOptions,
+  );
```
[the remark this answers is not in the document]
Comment (jessevanmuijden) [reply packages/tools/kadena-cli/src/account/accountHelpers.ts:54]: https://app.asana.com/0/1205969628270714/1206113181762573/f

File: packages/tools/kadena-cli/src/account/getBalanceCommand.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        config.accountConfig.account,
+        config.networkConfig.networkId,
+        config.chainId,
+        config.networkConfig.networkHost,
+      );
+      console.log(chalk.green(`\nThe balance of ${config.account} on chain ${config.chainId} on ${config.network} is: ${balance}.\n`));
+    } catch (e) {
+      console.log(chalk.red(e.message));
+    }
+  },
```
[the remark this answers is not in the document]
Comment (jessevanmuijden) [reply packages/tools/kadena-cli/src/account/getBalanceCommand.ts:23]: https://app.asana.com/0/1205969628270714/1206113181762565/f

File: packages/tools/kadena-cli/src/config/initConfigCommand.ts [hunk 4]
```diff
@@ -1,153 +1,20 @@
...
+      await import('./../networks/init.js');
+      console.log(chalk.green('Configured default networks.'));
 
-      // TODO: make this fix nicer
-      await import('./../utils/bootstrap.js');
+      await import('./../devnet/init.js');
+      console.log(chalk.green('Configured default devnets.'));
 
-      await runConfigInitialization(program, version, args);
+      console.log(chalk.green('Configuration complete!'));
```
[the remark this answers is not in the document]
Comment (jessevanmuijden) [reply packages/tools/kadena-cli/src/config/initConfigCommand.ts:18]: https://app.asana.com/0/1205969628270714/1206113181762573/f

File: packages/tools/kadena-cli/src/constants/devnets.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    port: 8080,
+    useVolume: false,
+    mountPactFolder: '',
+    version: 'latest',
+  },
+};
+
+export const defaultDevnetsPath: string = `${process.cwd()}/.kadena/devnets`;
+export const standardDevnets: string[] = ['devnet'];
+export const defaultDevnet: string = 'devnet';
```
[the remark this answers is not in the document]
Comment (jessevanmuijden) [reply packages/tools/kadena-cli/src/constants/devnets.ts:23]: https://app.asana.com/0/1205969628270714/1206113181762573/f

File: packages/tools/kadena-cli/src/constants/keysets.ts [hunk 6]
```diff
@@ -0,0 +1,10 @@
+export const keysetDefaults = {
+  sender00: {
+    name: 'sender00',
+    predicate: 'keys-all',
+    publicKeysFromKeypairs: ['sender00'],
+    publicKeys: '',
+  }
+};
+
+export const defaultKeysetsPath: string = `${process.cwd()}/.kadena/keysets`;
```
[the remark this answers is not in the document]
Comment (jessevanmuijden) [reply packages/tools/kadena-cli/src/constants/keysets.ts:10]: https://app.asana.com/0/1205969628270714/1206113181762573/f

File: packages/tools/kadena-cli/src/devnet/devnetHelpers.ts [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+export function displayDevnetConfig(
+  devnetConfig: IDevnetsCreateOptions,
+): void {
+  const log = console.log;
+  const formatLength = 80; // Maximum width for the display
+
+  const displaySeparator = (): void => {
+    log(chalk.green('-'.padEnd(formatLength, '-')));
+  };
+
```
[the remark this answers is not in the document]
Comment (jessevanmuijden) [reply packages/tools/kadena-cli/src/devnet/devnetHelpers.ts:117]: https://app.asana.com/0/1205969628270714/1206113181762565/f

File: packages/tools/kadena-cli/src/devnet/devnetHelpers.ts [hunk 8]
```diff
@@ -0,0 +1,205 @@
...
+    const valueDisplay =
+      (value?.trim() ?? '') !== '' ? chalk.green(value!) : chalk.red('Not Set');
+
+    const defaultIndicator =
+      isDefault === true ? chalk.yellow(' (Using defaults)') : '';
+    const keyValue = `${key}: ${valueDisplay}${defaultIndicator}`;
+    const remainingWidth =
+      formatLength - keyValue.length > 0 ? formatLength - keyValue.length : 0;
+    return `  ${keyValue}${' '.repeat(remainingWidth)}  `;
+  };
```
[the remark this answers is not in the document]
Comment (jessevanmuijden) [reply packages/tools/kadena-cli/src/devnet/devnetHelpers.ts:177]: https://app.asana.com/0/1205969628270714/1206113181762565/f

**Pull request closed without merging**